rustup-macros/src/lib.rs

```diff
@@ -80,8 +79,10 @@ pub fn unit_test(
 
 fn test_inner(mod_path: String, mut input: ItemFn) -> syn::Result<TokenStream> {
     if input.sig.asyncness.is_some() {
-        let before_ident = format_ident!("{}::before_test_async", mod_path);
-        let after_ident = format_ident!("{}::after_test_async", mod_path);
+        let before_ident = format!("{}::before_test_async", mod_path);
+        let before_ident = syn::parse_str::<Expr>(&before_ident)?;
```

Comment:
Shouldn't that be a `Path` rather than an `Expr`? If not, maybe give an example?

Maybe use inline format args?

Reply:
Expr is what works below - this part of rust isn't my forte, so if you wanted to muck around and make it more beautiful I suspect you'd get better code than I've written :).

Yes, it could use inline formatting, the format_ident macro didn't from memory, and I didn't think to change when changing.

That said there is more in the file that this is consistent with, so I'm inclined to come back later, and update the whole file.

Reply:
Fair enough.